test/helper/helper.ts

```diff
@@ -49,3 +50,37 @@ export function createEmitterAndSpy(
       emitter.emit(fn ? fn(event) : event.type),
   ];
 }
+
+/**
+ * TextView emulates an external editor like CodeMirror to test whether change
+ * events are delivered properly.
+ */
+export class TextView {
+  private value: string;
+
+  constructor() {
+    this.value = '';
+  }
+
+  public applyChanges(changes: Array<Change>): void {
+    // const oldValue = this.value;
+    const changeLogs = [];
+    for (const change of changes) {
+      if (change.type === ChangeType.Content) {
+        this.value = [
+          this.value.substring(0, change.from),
+          change.content,
+          this.value.substring(change.to),
+        ].join('');
+        changeLogs.push(
+          `{f:${change.from}, t:${change.to}, c:${change.content || ''}}`,
+        );
+      }
+    }
+    // console.log(`apply: ${oldValue}->${this.value} [${changeLogs.join(',')}]`);
```

Comment:
I think we can remove this command 🤔

Reply:
Thanks for the comment. The log was useful for debugging, so I extracted it as an option.